pkg/yamlmeta/document.go

```diff
@@ -4,27 +4,42 @@
 package yamlmeta
 
 import (
+	"fmt"
+	"strings"
+
+	"github.com/k14s/ytt/pkg/filepos"
 	"github.com/k14s/ytt/pkg/yamlmeta/internal/yaml.v2"
 )
 
-func (n *Document) IsEmpty() bool {
-	if n.Value == nil {
+func (d *Document) IsEmpty() bool {
+	if d.Value == nil {
 		return true
 	}
 	// TODO remove doc empty checks for map and array
-	if typedMap, isMap := n.Value.(*Map); isMap {
+	if typedMap, isMap := d.Value.(*Map); isMap {
 		return len(typedMap.Items) == 0
 	}
-	if typedArray, isArray := n.Value.(*Array); isArray {
+	if typedArray, isArray := d.Value.(*Array); isArray {
 		return len(typedArray.Items) == 0
 	}
 	return false
 }
 
-func (n *Document) AsYAMLBytes() ([]byte, error) {
-	return yaml.Marshal(convertToLowYAML(convertToGo(n.Value)))
+func (d *Document) RawDataAtLine(position *filepos.Position) (string, error) {
```

Comment:
it would be good to figure out how to take advantage of existing infrastructure around correlating lines to content: https://github.com/vmware-tanzu/carvel-ytt/blob/e24944a5abee281783e1ad62a483d239a80bcc04/pkg/template/compiled_template_error.go#L183.

Reply:
Position now has this responsibility. parse sets it up and the error messages can use it without needing to ask the Document for the content at that line.